enterprise_access/apps/customer_billing/models.py

```diff
@@ -731,20 +731,17 @@ def populate_with_summary_data(self):
 
         # Get subscription plan UUID from related workflow
         if stripe_event_data.checkout_intent and stripe_event_data.checkout_intent.workflow:
-            try:
-                # Fetch model from the Django app registry to avoid
-                # a circular import.
-                subs_output_model = apps.get_model(
-                    'provisioning', 'GetCreateSubscriptionPlanStepOutput',
-                )
-                plan_output = subs_output_model.objects.filter(
-                    workflow=stripe_event_data.checkout_intent.workflow
-                ).first()
-
-                if plan_output and plan_output.output_object:
-                    self.subscription_plan_uuid = plan_output.output_object.get('subscription_plan_uuid')
-            except ImportError:
-                logger.warning("Could not import GetCreateSubscriptionPlanStepOutput")
+            # Fetch model from the Django app registry to avoid
+            # a circular import.
+            subscription_step_model = apps.get_model(
+                'provisioning', 'GetCreateSubscriptionPlanStep',
+            )
+            subscription_step = subscription_step_model.objects.filter(
+                workflow_record_uuid=stripe_event_data.checkout_intent.workflow.uuid,
+            ).first()
+
+            if subscription_step and subscription_step.output_object:
```

Comment:
[nit/non-blocking] Super nit, but,
`if subscription_step and subscription_step_output := subscription_step.output_object:`